### PR TITLE
Added AVPlayer rate observing

### PR DESCRIPTION
**Pull request by pavliuko**
@aol-public/mobile-sdk-team: Ready for review.

Related to [PR](https://github.com/vidible/mobile-sdk-ios/pull/569)

There is a need to observe changing of `AVPlayer.rate` changes for tvOS, because of play/pause button on remote control changes rate of active `AVPlayer` under the hood.

### CODE REVIEW

**Review by oleksii-demedetskyi — DISMISSED** [review]
Add a description of what you want to achieve with this PR, why we need it, etc.

**Comment by pavliuko** [comment]
@AlexeyDemedetskiy Description extended.

**Comment by AndriiMoskvin** [comment]
@OleksandrPavliuk Don't understand why we need `AVPlayer.rate` observing.

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]
@OleksandrPavliuk Don't understand why we need AVPlayer.rate observing.

**Comment by pavliuko** [comment]
@AndriiMoskvin as i mentioned [here](https://github.com/vidible/mobile-sdk-ios/pull/569) 
We need it to track play/pause on tvOS

**Comment by pavliuko** [comment]
@AndriiMoskvin @AlexeyDemedetskiy please take a look ones again

File: VideoRenderer/VideoRenderer/SystemPlayerObserver.swift [hunk 1]
```diff
@@ -171,7 +181,7 @@ public final class SystemPlayerObserver: NSObject {
             weak var this = self
             func emitDidChangeRate(for timebase: CMTimebase) {
```
Comment (AndriiMoskvin) [inline VideoRenderer/VideoRenderer/SystemPlayerObserver.swift]: `emitDidChangeRate` -> `emitDidChangeTimebaseRate `

File: VideoRenderer/VideoRenderer/SystemPlayerObserver.swift [hunk 2]
```diff
@@ -67,6 +73,10 @@ public final class SystemPlayerObserver: NSObject {
         }
         
         switch keyPath {
+        case #keyPath(AVPlayer.rate):
+            if let newItem = newValue() as Float? {
```
Comment (AndriiMoskvin) [inline VideoRenderer/VideoRenderer/SystemPlayerObserver.swift]: `guard` here is preferred.

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]

File: VideoRenderer/VideoRenderer/SystemPlayerObserver.swift (hunk 2 above)
Comment (pavliuko) [reply VideoRenderer/VideoRenderer/SystemPlayerObserver.swift]: Adjusted.

File: VideoRenderer/VideoRenderer/SystemPlayerObserver.swift (hunk 1 above)
Comment (pavliuko) [reply VideoRenderer/VideoRenderer/SystemPlayerObserver.swift]: Adjusted.

File: VideoRenderer/VideoRenderer/SystemPlayerObserver.swift [hunk 3]
```diff
@@ -67,6 +73,9 @@ public final class SystemPlayerObserver: NSObject {
         }
         
         switch keyPath {
+        case #keyPath(AVPlayer.rate):
+            guard let newItem = newValue() as Float? else { return }
+            self.emit(.didChangeRate(to: newItem))
```
Comment (AndriiMoskvin) [inline VideoRenderer/VideoRenderer/SystemPlayerObserver.swift]: `self` could be dropped.

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]

File: VideoRenderer/VideoRenderer/SystemPlayerObserver.swift (hunk 3 above)
Comment (pavliuko) [reply VideoRenderer/VideoRenderer/SystemPlayerObserver.swift]: Done.

**Review by AndriiMoskvin — APPROVED** [review]

**Pull request merged**